go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/spacemeshos/poet
 
-go 1.21
+go 1.21.0
```

Comment:
Why not?
```suggestion
go 1.21.1
```

Reply:
I updated the version in `go.mod`.

The version specified here only says what is the **minimum** version required to build the module, the CI should use the newest 1.21.x anyway. `govulncheck` reports what I think is a false positive: It says that `poet.poetMain` indirectly calls `html/template.Template.Execute` which I don't think is true 🤔 but had a bug that was fixed in `go 1.21.1` so I updated it to be sure.